hoomd/md/tune/nlist_buffer.py

```diff
@@ -0,0 +1,213 @@
+# Copyright (c) 2009-2022 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+"""Provide a tuner for `hoomd.md.nlist.NeighborList.buffer`."""
+
+import copy
+
+import numpy as np
+
+import hoomd.custom
+import hoomd.data
+from hoomd.data.typeconverter import OnlyTypes, SetOnce
+import hoomd.logging
+import hoomd.tune
+import hoomd.trigger
+from hoomd.md.nlist import NeighborList
+
+
+class _IntervalTPS:
+
+    def __init__(self, simulation):
+        self._simulation = simulation
+        self._last_timestep = simulation.timestep
+        self._last_walltime = simulation.walltime
```

Comment:
```suggestion
        self._last_timestep = None
        self._last_walltime = None
```
When the tuner is added between two calls to `run`, this records the final walltime of the previous `run` which is not a valid start time for the next run.

To fully fix this problem, also make this change to `System.cc`
```
--- a/hoomd/System.cc
+++ b/hoomd/System.cc
@@ -84,6 +84,8 @@ void System::run(uint64_t nsteps, bool write_at_start)
 
     // initialize the last status time
     m_initial_time = m_clk.getTime();
+    m_last_walltime = 0;
+    m_last_TPS = 0;
```

Reply:
Something else might need to be changed. I implemented these locally and sometimes the tuner gets stuck near the initial buffer value and doesn't move much from it.